main/src/cgeo/geocaching/export/PersonalnoteExport.java

```diff
@@ -0,0 +1,84 @@
+package cgeo.geocaching.export;
+
+import cgeo.geocaching.CgeoApplication;
+import cgeo.geocaching.R;
+import cgeo.geocaching.activity.ActivityMixin;
+import cgeo.geocaching.connector.ConnectorFactory;
+import cgeo.geocaching.connector.IConnector;
+import cgeo.geocaching.connector.capability.PersonalNoteCapability;
+import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.utils.AsyncTaskWithProgress;
+import cgeo.geocaching.utils.Log;
+
+import android.app.Activity;
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Exports personal notes.
+ */
+public class PersonalnoteExport extends AbstractExport {
+
+    private int persNotesCount = 0;
+
+    public PersonalnoteExport() {
+        super(R.string.export_persnotes);
+    }
+
+    @Override
+    public void export(@NonNull final List<Geocache> cachesList, @Nullable final Activity activity) {
+        final Geocache[] caches = cachesList.toArray(new Geocache[cachesList.size()]);
+        new ExportTask(activity).execute(caches);
+    }
+
+    private class ExportTask extends AsyncTaskWithProgress<Geocache, Boolean> {
+        /**
+         * Instantiates and configures the task for exporting personal notes.
+         */
+        ExportTask(@Nullable final Activity activity) {
+            super(activity, getProgressTitle(), CgeoApplication.getInstance().getString(R.string.export_persnotes), true);
+        }
+
+        @Override
+        protected Boolean doInBackgroundInternal(final Geocache[] caches) {
+            try {
+                for (final Geocache cache : caches) {
+                    final IConnector connector = ConnectorFactory.getConnector(cache);
+                    if (connector instanceof PersonalNoteCapability && StringUtils.isNotBlank(cache.getPersonalNote())) {
+                        publishProgress(++persNotesCount);
```

Comment:
I didn't notice this on first review: I suggest to publish the progress outside the if. If you have a mixed collection of caches where some do not have personal note capability, the progress would not report these caches, and when it has finished the loop, the progress bar would suddenly jump to the end (e.g. if you have 300 caches with personal note capability and 100 without, the progress would go smoothly to 75% and then suddenly finish).
If progress is always reported, then the progress bar may do small jumps in between, but it will better reflect the overall progress.